lib/ts/recipe/recipeRoutes/index.tsx

```diff
@@ -0,0 +1,130 @@
+import React from "react";
+import NormalisedURLPath from "supertokens-web-js/lib/build/normalisedURLPath";
+import { RoutingComponent } from "../../components/routingComponent";
+import { getSuperTokensRoutesForReactRouterDom } from "../../components/superTokensRoute";
+import { getSuperTokensRoutesForReactRouterDomV6 } from "../../components/superTokensRouteV6";
+import { BaseFeatureComponentMap, ComponentWithRecipeAndMatchingMethod } from "../../types";
+import { getCurrentNormalisedUrlPath } from "../../utils";
+import { RecipeFeatureComponentMap } from "../../types";
+
+export abstract class RecipeRoutes {
+    private pathsToFeatureComponentWithRecipeIdMap?: BaseFeatureComponentMap;
+
+    private static reactRouterDom?: { router: { Route: any }; useHistoryCustom: () => any };
+    private static reactRouterDomIsV6: boolean | undefined = undefined;
+
+    static getRoutes(reactRouterDom: any, instance?: RecipeRoutes): JSX.Element[] {
+        if (instance === undefined) {
+            throw new Error(
+                // eslint-disable-next-line @typescript-eslint/quotes
+                "Instance should be passed"
+            );
+        }
+        if (reactRouterDom === undefined) {
+            throw new Error(
+                // eslint-disable-next-line @typescript-eslint/quotes
+                'Please use getSuperTokensRoutesForReactRouterDom like getSuperTokensRoutesForReactRouterDom(require("react-router-dom")) in your render function'
+            );
+        }
+        RecipeRoutes.reactRouterDom = reactRouterDom;
```

Comment:
We could also check for equality if `RecipeRoutes.reactRouterDom` already has a value. We didn't need this previously because in almost all cases this would only be called once, but this will change now.

Reply:
This hasn't happened